features/126-record-id/api.md

```diff
@@ -0,0 +1,73 @@
+# Removing `type/` from APIs
+
+## Overview of changes
+
+This document propose what changes are needed to remove `type/` from Skygear API
+so that developer will have consistent experience when needing to specify
+a Record ID.
+
+Throughout this document the different ID formats are described as:
+
+Deprecated ID - "note/0ebd484c-ed82-4b04-aba7-33fc530f3c88"
+Record Type - "note"
+Record ID - "0ebd484c-ed82-4b04-aba7-33fc530f3c88"
+
+### Changes to SDK API
+
+* API that accepts deprecated ID as parameter should accept a Record Type and
+  a Record ID as parameter, both of `string` type.
+* API that accepts a list of deprecated IDs as parameter should accept a Record
+  Type (string) and a list of Record ID (string).
+* API that returns deprecated IDs (or list) should return Record ID (string),
+  the Record Type is either
+  * Will be omitted if Record Type is already known. For example, delete
+    operation will return a Record IDs, and the Record Type is inferred from
+    the Record Type used in the delete operation.
+  * Will return a Reference instead, which has separate Record ID and Record
+    Type fields.
+  * Other cases will need to be discussed.
+* API that supports deprecated IDs with multiple Record Types will no longer
```

Comment:
just to confirm the only APIs fall into this category are `fetch` and `delete`

Reply:
Yes.